File: testdata/fixtures/bench/go/vulnerable/a01_path_traversal.go
// Source: CWE-22 - Path Traversal in Go file server
// Expected: BATOU-TRV-001 (Path Traversal via user input in file path)
// OWASP: A01:2021 - Broken Access Control (Path Traversal)

package handler

import (
	"io"
	"net/http"
	"os"
	"path/filepath"
)

const uploadDir = "./uploads"

func ServeFile(w http.ResponseWriter, r *http.Request) {
	filename := r.URL.Query().Get("file")
	fullPath := filepath.Join(uploadDir, filename)
	file, err := os.Open(fullPath)
	if err != nil {
		http.Error(w, "File not found", http.StatusNotFound)
		return
	}
	defer file.Close()
	io.Copy(w, file)
}

func DownloadAttachment(w http.ResponseWriter, r *http.Request) {
	name := r.URL.Query().Get("name")
	path := "/var/attachments/" + name
	http.ServeFile(w, r, path)
}

File: testdata/fixtures/bench/lua/vulnerable/a01_path_traversal.lua
-- Source: CWE-22 - Path Traversal via io.open in Lua
-- Expected: BATOU-LUA
-- OWASP: A01:2021 - Broken Access Control (Path Traversal)

local input = arg[1]
local path = "/var/uploads/" .. input
local file = io.open(path, "r")
if file then
    local content = file:read("*a")
    print(content)
    file:close()
end
